specs/agents/tracing-spans-destination.md

```diff
@@ -84,6 +84,8 @@ string through API, agents MUST infer the value of this field based on propertie
 If no value is set to the `context.destination.service.resource` field, the logic for automatically inferring
 it MUST be the following:
 
+Q3: Though above it is says "value should be inferred from `context.service.target.*` fields", this pseudo-code does not consider service.target. Should this be updated to calculate from service.target?
+
```

Comment:
Q3: The Java implementation has `ServiceTarget.getDestinationResource()` for determining `destination.service.resource` from `service.target`, and it uses `.withNameOnlyDestinationResource()` calls to handle the special case of *not* including using a `${service.target.type}/` prefix on the value.

The Python implementation currently is still using this current logic for determining `destination.service.resource` and, IIUC, it will not consider a manually set value of `service.target.type` when setting `destination.service.resource`.

Should this section be updated to calculate from service.target and other span fields?  I was playing around with what the logic should be and so far I have this:


```js
if (!context.service?.target) {
    // If there is no `service.target`, e.g. if `span.setServiceTarget(null,
    // null)` was called by the user, then do not set a
    // `destination.service.resource`.
} else {
    context.destination.service = {
        // These two could be skipped if the upstream APM server is known to
        // be >=7.14.
        type: '',
        name: ''
    }

    if (!context.service.target.name) {
        // Note: This `!...` checks if the value is undefined, null, or the empty string.
        // If we only have `.type`, then use that.
        context.destination.service.resource = context.service.target.type
    } else if (!context.service.target.type) {
        // If we only have `.name`, then use that.
        context.destination.service.resource = context.service.target.name
    } else if (!context.db && !context.message && context.http?.url) {
        // If this is not a DB or messaging span, and we have an `http.url`,
        // then do *not* put `.type` in the resource string.
        context.destination.service.resource = context.service.target.name
    } else {
        context.destination.service.resource = `${context.service.target.type}/${context.service.target.name}`
    }
}
```

Again, as with Q2, another option would be to check `span.type === "db" | "messaging"` rather than checking for `context.{db.message}.` values.

Thoughts?

Reply:
On the user-side, we can expect to have either `context.destination.resource` OR ( `service.target.type` + `service.target.name`), trying to set all values would definitely be confusing and complex (having to deal with setting one or the other in any order for example). For Java, the `withNameOnlyDestinationResource` method is only meant to be used for specific instrumentations (for example HTTP clients) that require it.

Reply:
> On the user-side, we can expect to have either `context.destination.resource` OR ( `service.target.type` + `service.target.name`) ...

I'm not sure what you mean here. What is the "user-side" here?

I was proposing to change the pseudo-code at https://github.com/elastic/apm/blob/main/specs/agents/tracing-spans-destination.md#destination-resource to what I've shown above, so that `context.destination.service.resource` is calculated/inferred from `context.service.target.*`.

Reply:
I updated this logic in commit 80f3782 to what I have in my Node.js agent impl here: https://github.com/elastic/apm-agent-nodejs/blob/a10bccfe797577f8414c957aaf1ec50ea581e8b9/lib/instrumentation/span.js#L146-L173